server/server.go

```diff
@@ -0,0 +1,128 @@
+// main package to be able to run the StudentsServer for now
+package main
+
+import (
+	"context"
+	"net"
+
+	spb "github.com/BetterGR/students-microservice/students_protobuf"
+	"google.golang.org/grpc"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	// define port
+	address            = "localhost:50052"
+	connectionProtocol = "tcp"
+)
+
+// StudentsServer is an implementation of GRPC Students microservice.
+type StudentsServer struct {
+	// throws unimplemented error
+	spb.UnimplementedStudentsServiceServer
+}
+
+// GetStudent search for the Student that corresponds to the given id and returns them.
+func (s *StudentsServer) GetStudent(ctx context.Context,
+	req *spb.GetStudentRequest) (*spb.GetStudentResponse, error) {
+	logger := klog.FromContext(ctx)
+	logger.Info("Received GetStudent request for:", "student_id", req.Id)
+
+	student := &spb.Student{
+		FirstName:  "Rick",
+		SecondName: "Roll",
+		Id:         req.Id,
+	}
+
+	return &spb.GetStudentResponse{
+		Student: student,
+	}, nil
+}
+
+// CreateStudent creates a new Student with the given details and returns them.
+func (s *StudentsServer) CreateStudent(ctx context.Context,
+	req *spb.CreateStudentRequest) (*spb.CreateStudentResponse, error) {
+	logger := klog.FromContext(ctx)
+	logger.Info("Received CreateStudent request for:",
+		"student_firstName", req.Student.FirstName, "student_secondName", req.Student.SecondName)
+
+	return &spb.CreateStudentResponse{Student: req.Student}, nil
+}
+
+// UpdateStudent updates the given Student and returns them after the update.
+func (s *StudentsServer) UpdateStudent(ctx context.Context,
+	req *spb.UpdateStudentRequest) (*spb.UpdateStudentResponse, error) {
+	logger := klog.FromContext(ctx)
+	logger.Info("Received UpdateStudent request for:",
+		"student_firstName", req.Student.FirstName, "student_secondName", req.Student.SecondName)
+
+	return &spb.UpdateStudentResponse{Student: req.Student}, nil
+}
+
+// GetStudentCourses searches the courses that the Student is enrolled in during the given semester and returns them.
+func (s *StudentsServer) GetStudentCourses(ctx context.Context,
+	req *spb.GetStudentCoursesRequest) (*spb.GetStudentCoursesResponse, error) {
+	logger := klog.FromContext(ctx)
+	logger.Info("Received GetStudentCourses request for:",
```

Comment:
Same. Considering the amount of logging these logs add, per request, it might make sense to increase the verbosity level. 

I think it makes sense to log on TRACE level, which is V=6. 5 would do as well.

Not a blocker.